lib/ex_doc.ex

```diff
@@ -61,8 +64,16 @@ defmodule ExDoc do
   # Helpers
 
   defp normalize_options(options) do
+    options_norm = []
+
     pattern = options[:source_url_pattern] || guess_url(options[:source_url], options[:source_ref] || "master")
-    Keyword.put(options, :source_url_pattern, pattern)
+    options_norm = options_norm ++ [source_url_pattern: pattern]
```

Comment:
Why you are not using `Keyword.put`? It will be faster and semantically correct.